src/EPR.Accreditation.Facade/Controllers/AccreditationController.cs

```diff
@@ -210,5 +210,40 @@ public async Task<IActionResult> GetLegalDocumentsAddress(Guid id)
 
             return Ok();
         }
+
+        /// <summary>
+        /// Updates the the reference number for the accrediation.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="referenceNumber"></param>
+        /// <returns></returns>
+        [HttpPut("ReferenceNumber")]
+        public async Task<IActionResult> UpdateReferenceNumber(
+            Guid id,
+            [FromBody] string referenceNumber)
+        {
+            await _accreditationService.UpdateReferenceNumber(id, referenceNumber);
+
+            return Ok();
+        }
+
+        [HttpGet("ReferenceNumber")]
+        public async Task<IActionResult> GetReferenceNumber(
+            Guid id)
+        {
+            var accreditation = await _accreditationService.GetAccrediation(id);
+
+            return Ok(accreditation.ReferenceNumber);
+        }
+
+
+        [HttpGet("Accreditation")]
+        public async Task<IActionResult> GetAccredition(
+            Guid id)
+        {
+            var accreditation = await _accreditationService.GetAccrediation(id);
```

Comment:
Do you need to pass an entire accreditation class back to Portal? This is the same DTO as was passed from API to Facade. We should really be passing only the properties we need in Portal. Therefore there should be a seperate DTO class under the Portal folder in DTOs, to help create that seperation